sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetCachedBatchSerializer.scala

```diff
@@ -1096,14 +1096,17 @@ protected class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer wi
      * is called on this iterator
      */
     class ColumnarBatchToCachedBatchIterator extends InternalRowToCachedBatchIterator {
+      // Exposing this just for tests to close the batch
+      var hostBatch: ColumnarBatch = null
```

Comment:
A cleaner way to handle this is to have `ColumnarBatchToCachedBatchIterator` implement `AutoCloseable` so the calling code can call `close()` to clean up anything that needs to be cleaned rather than reaching into the internals.

Reply:
But who will call close? We are returning this `ColumnarBatchToCachedBatchIterator` to `InMemoryRelation` which is calling this from `buildBuffers`. Only once the persistence is done is when we can possibly close the batch and there is no callback for that right now.

Reply:
The close would be called by the testing code, just like it's trying to close the hostBatch now.  It's avoiding the need to expose internals on how to cleanup and then make the test know more than it should about how to cleanup the instance.

Reply:
I wasn't referring to test. but anyways, I finally understand what you are saying. Will push an update